### ISSUE TITLE
[Bug]: FluentUI ContextualMenu Accessibility Issue nested-interactive

### ISSUE THREAD

**Issue by mzavala722** [microsoft/fluentui#27595]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
Package version: @fluentui/react: 8.93.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/contextualmenu

## Repro Steps
- Navigate to the documentation for the ContextualMenu component: https://developer.microsoft.com/en-us/fluentui#/controls/web/contextualmenu
- Find the "ContextualMenu with checkable menu items and toggleable split button" example
- Open the contextual menu
- Run the Accessibility Insights for Web fast pass.

## Actual Behavior
Implementations of the FluentUI ContextualMenu are currently failing the Accessibility Insights for Web fast pass. See screenshots and attached logs.

![image](https://user-images.githubusercontent.com/130993509/232533646-f2a74a3e-5922-401d-8e1b-763583d2ffd6.png)
![image](https://user-images.githubusercontent.com/130993509/232540750-8d55b57f-3f6d-47a6-833f-f95fd87da844.png)

## Expected Behavior
The ContextualMenu component should pass the Accessibility Insights for Web fast pass without any accessibility issues.

### Logs

```shell
Title: WCAG 4.1.2: Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies (.splitContainer-422[aria-posinset="6"][role="menuitemcheckbox"])
Tags: Accessibility, WCAG 4.1.2, nested-interactive

Issue: Ensures interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies (nested-interactive - https://dequeuniversity.com/rules/axe/4.6/nested-interactive?application=msftAI)

Target application: Fluent UI - Controls - React - ContextualMenu - https://developer.microsoft.com/en-us/fluentui#/controls/web/contextualmenu

Element path: .splitContainer-422[aria-posinset="6"][role="menuitemcheckbox"]

Snippet: <div role="menuitemcheckbox" class="splitContainer-422" aria-disabled="false" aria-expanded="false" aria-haspopup="true" aria-posinset="6" aria-setsize="10" tabindex="-1" data-is-focusable="true">

How to fix: 
Fix any of the following:
  Using a negative tabindex on an element inside an interactive control does not prevent assistive technologies from focusing the element (even with 'aria-hidden=true')

Environment: Microsoft Edge version 112.0.1722.39

====

This accessibility issue was found using Accessibility Insights for Web 2.38.1 (axe-core 4.6.3), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.
```

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by smhigley** [comment]
Hi @mzavala722 this is a somewhat complex issue, but the reason we've done this is because the v8 SplitButton is two separate controls functioning as a single tab stop. This is not a pattern that supports good semantic and keyboard accessibility, but it would be a breaking change to alter it in v8. We have changed splitbutton in v9 to be two separate tab stops, and this is no longer an issue there.

The reason for the nested interactives is to provide the best possible practical accessibility in v8 without changing the tabbing behavior. We have a note about it on our docs site under the "Accessibility" heading for Button: https://developer.microsoft.com/en-us/fluentui#/controls/web/button
